src/components/structures/RoomView.tsx

```diff
@@ -809,7 +810,21 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
         return this.state.room?.roomId ?? this.state.roomId;
     };
 
-    private getPermalinkCreatorForRoom(room: Room): RoomPermalinkCreator {
+    private getPermalinkCreatorForRoom(): RoomPermalinkCreator {
+        const { room, roomId } = this.state;
+
+        // if room is undefined, attempt to use the roomId to create and store a permalinkCreator
+        if (room === undefined) {
+            if (isNotUndefined(roomId)) {
+                const permalinkCreator = new RoomPermalinkCreator(null, roomId);
+                this.permalinkCreators[roomId] = permalinkCreator;
+                return permalinkCreator;
+            } else {
+                // this will throw every time, should we throw here instead?
+                return new RoomPermalinkCreator(null, null);
+            }
+        }
+
```

Comment:
Looks like you have a missing closing brace


```suggestion
}
```

Reply:
...not quite sure how I missed that, cheers